File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/AudioTypes.java
package com.merseyside.filemanager.mime_types;


import android.mtp.MtpConstants;

public enum AudioTypes {

    MP3  (new MimeType(getFormat(), "mpeg", MtpConstants.FORMAT_MP3)),
    MPGA (new MimeType(getFormat(), "mpeg", MtpConstants.FORMAT_MP3)),
    M4A  (new MimeType(getFormat(), "m4a",  MtpConstants.FORMAT_MPEG)),
    WAV  (new MimeType(getFormat(), "x-wav",MtpConstants.FORMAT_WAV)),
    WMA  (new MimeType(getFormat(), "mpeg", MtpConstants.FORMAT_MP3)),
    OGG  (new MimeType(getFormat(), "ogg",  MtpConstants.FORMAT_OGG)),
    AAC  (new MimeType(getFormat(), "aac",  MtpConstants.FORMAT_AAC)),
    FLAC (new MimeType(getFormat(), "flac", MtpConstants.FORMAT_FLAC));

    MimeType type;

    AudioTypes(MimeType type) {
        this.type = type;
    }

    public MimeType getValue() {
        return type;
    }

    public static String getFormat() {
        return "audio";
    }
}

File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/AudioTypes.kt
package com.merseyside.filemanager.mime_types

import android.mtp.MtpConstants

enum class AudioTypes(var value: MimeType) {
    MP3(
        MimeType(format, "mpeg", MtpConstants.FORMAT_MP3)
    ),
    MPGA(
        MimeType(
            format,
            "mpeg",
            MtpConstants.FORMAT_MP3
        )
    ),
    M4A(
        MimeType(format, "m4a", MtpConstants.FORMAT_MPEG)
    ),
    WAV(
        MimeType(format, "x-wav", MtpConstants.FORMAT_WAV)
    ),
    WMA(
        MimeType(
            format,
            "mpeg",
            MtpConstants.FORMAT_MP3
        )
    ),
    OGG(
        MimeType(format, "ogg", MtpConstants.FORMAT_OGG)
    ),
    AAC(
        MimeType(format, "aac", MtpConstants.FORMAT_AAC)
    ),
    FLAC(
        MimeType(
            format,
            "flac",
            MtpConstants.FORMAT_FLAC
        )
    );

    companion object {
        val format: String
            get() = "audio"
    }
}